cloudnet-cord/cloudnet-setup/src/main/java/de/dytanic/cloudnet/setup/spigot/SpigotBuilder.java

```diff
@@ -0,0 +1,200 @@
+import org.jsoup.nodes.Element;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+public final class SpigotBuilder {
+
+    private final static String buildToolsUrl = "https://hub.spigotmc.org/jenkins/job/BuildTools/lastSuccessfulBuild/artifact/target/BuildTools.jar";
+    private final static String versionsUrl = "https://hub.spigotmc.org/versions/";
+    private static Process exec;
+
+    /**
+     * Start the process to choice the Spigot version And build after choice
+     *
+     * @param reader to read the answer
+     */
+    public static void start(final ConsoleReader reader) {
+        System.out.println("Fetch Spigot Versions");
+        List<String> versions = loadVersions();
+        System.out.println("Available Spigot Versions");
+        System.out
+                .println("-----------------------------------------------------------------------------");
+        System.out.println("Spigot Version\n");
+        System.out
+                .println("-----------------------------------------------------------------------------");
+        versions.forEach(System.out::println);
+        System.out
+                .println("-----------------------------------------------------------------------------");
+        System.out.println("Please select a version to continue the install process");
+        String answer = null;
+        while (answer == null) {
+            String name = null;
+            try {
+                name = reader.readLine().toLowerCase();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            String finalAnswer = name;
+            if (versions.stream().anyMatch(e -> e.equalsIgnoreCase(finalAnswer))) {
+                answer = name;
+                buildSpigot(finalAnswer);
+            } else if (versions.stream().noneMatch(e -> e.equalsIgnoreCase(finalAnswer))) {
+                System.out.println("Version does not exist!");
+            }
+        }
+    }
+
+    /**
+     * @return A list for available versions of spigot
+     */
+    private static LinkedList<String> loadVersions() {
+        LinkedList<String> array = new LinkedList<>();
+        try {
+            Document doc = Jsoup.connect(versionsUrl).userAgent(
+                    "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.95 Safari/537.11")
+                    .get();
+            for (Element file : doc.select("a")) {
+                String rawName = file.attr("href");
+                if (rawName.contains("../")) {
+                    continue;
+                }
+                if (!(rawName.contains("latest.json") | rawName.startsWith("1."))) {
```

Comment:
```suggestion
                if (!(rawName.contains("latest.json") || rawName.startsWith("1."))) {
```

Please use logical expressions when dealing with booleans (short-circuiting)